v21.1/performance-recipes-solutions.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Performance tuning recipe solutions
+summary: Identifying possible fixes for common performance problems
+toc: true
+toc_not_nested: true
+---
+
+This page provides solutions for common performance issues in your clusters. See [the recipes](performance-recipes.html) to identify performance problems in your workload.
+
+<div class="filters clearfix">
+    <button class="filter-button page-level" data-scope="contention"><strong>Contention</strong></button>
+    <button class="filter-button page-level" data-scope="fullscans">Full table scans<strong></strong></button>
+    <button class="filter-button page-level" data-scope="primarykey">Suboptimal primary keys<strong></strong></button>
+</div>
+
+<section class="filter-content" markdown="1" data-scope="contention">
+## Indicators that your workoad is experiencing contention
+
+Your application is experiencing degraded performance with serialization errors like:
+
+  - `SQLSTATE: 40001`
+  - `RETRY_WRITE_TOO_OLD`
+  - `RETRY_SERIALIZABLE`
+
+The [SQL Statement Contention graph](ui-sql-dashboard.html#sql-statement-contention) graph is showing spikes over time.
+
+{%comment%} Add screenshot of high-contention graph {%endcomment%}
+
+The [SQL Statement Errors graph](ui-sql-dashboard.html#sql-statement-errors) graph is showing spikes in retries over time.
+
+{%comment%} Add screenshot of high-retries graph {%endcomment%}
+
+## Fixing contention problems
+
+{% include {{ page.version.version }}/performance/statement-contention.md %}
+</section>
+
+<section class="filter-content" markdown="1" data-scope="fullscans">
+
+## Indicators that your workload has statements with full table scans
+
+The following query returns statements with full table scans in their statement plan:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SHOW FULL TABLE SCANS;
+~~~
+
+The following query against the `CRDB_INTERNAL.node_statement_statistics` table returns results:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT count(*) as total_full_scans
+FROM crdb_internal.node_statement_statistics
+WHERE FullTableScan = 'True';
+~~~
+
+Viewing the statement plan on the [Statement details page](ui-statements-page.html#statement-details-page) of the DB Console indicates that the plan contains full table scans.
+
+If the statement plans returned by the [`EXPLAIN`](sql-tuning-with-explain.html) and [`EXPLAIN ANALYZE` commands](explain-analyze.html) indicate that there are full table scans.
+
```

Comment:
There's another signal/problem to include:

The Full Table/Index Scans chart in the DB console SQL metrics dashboard is showing spikes over time.